compiler/daml-lf-verify/src/DA/Daml/LF/Verify/Generate.hs

```diff
@@ -0,0 +1,192 @@
+        else expOut ^. goUpd
+  return $ set goUpd updSet
+         $ over (goDel . devars) ((:) (fst $ chcArgBinder cho))
+         expOut
+
+genTemplate :: MonadDelta m => Template -> m Delta
+genTemplate = undefined -- TODO
+
+genExpr :: MonadDelta m => Expr -> m GenOutput
+genExpr = \case
+  ETmApp fun arg  -> genForTmApp fun arg
+  ETyApp expr typ -> genForTyApp expr typ
+  EVar name       -> genForVar name
+  EVal w          -> genForVal w
+  EUpdate (UCreate tem arg)              -> genForCreate tem arg
+  EUpdate (UExercise tem ch cid par arg) -> genForExercise tem ch cid par arg
+  _ -> error "Not implemented"
+
+genForTmApp :: MonadDelta m => Expr -> Expr -> m GenOutput
+genForTmApp fun arg = do
+  funOut <- genExpr fun
+  argOut <- genExpr arg
+  case funOut ^. goExp of
+    ETmLam bndr body -> do
+      let updDelta = concatDelta (funOut ^. goDel) (argOut ^. goDel)
+          subst    = singleExprSubst (fst bndr) (argOut ^. goExp)
+          resExpr  = substituteTmTm subst body
+      resOut <- introDelta updDelta $ genExpr resExpr
+      return $ combineGO resOut
+             $ combineGO funOut argOut
+    fun'             -> return $ updateGOExpr (ETmApp fun' (argOut ^. goExp))
+                               $ combineGO funOut argOut
+
+genForTyApp :: MonadDelta m => Expr -> Type -> m GenOutput
+genForTyApp expr typ = do
+  exprOut <- genExpr expr
+  case exprOut ^. goExp of
+    ETyLam bndr body -> do
+      let subst   = singleTypeSubst (fst bndr) typ
+          resExpr = substituteTyTm subst body
+      resOut <- introDelta (exprOut ^. goDel) $ genExpr resExpr
+      return $ combineGO resOut exprOut
+    expr'            -> return $ updateGOExpr (ETyApp expr' typ) exprOut
+
+genForVar :: MonadDelta m => ExprVarName -> m GenOutput
+genForVar name = lookupDExprVar name
+                 >> return (GenOutput (EVar name) emptyUpdateSet emptyDelta)
+
+genForVal :: MonadDelta m => (Qualified ExprValName) -> m GenOutput
+genForVal w = lookupDVal w
+              >>= \ (expr, upds) -> return (GenOutput expr upds emptyDelta)
+
+genForCreate :: MonadDelta m => (Qualified TypeConName) -> Expr -> m GenOutput
+genForCreate tem arg = do
+  argOut <- genExpr arg
+  case argOut ^. goExp of
+    argExpr@(ERecCon _ fs) -> return (GenOutput (EUpdate (UCreate tem argExpr))
+                                      -- TODO: We could potentially filter here
+                                      -- to only store the interesting fields?
+                                      (set usCre [UpdCreate tem fs] emptyUpdateSet)
+                                      (argOut ^. goDel))
+    _                      -> throwError EEnumTypeWithParams
+    -- TODO: This is a random error, as we do not have access to the expected
+    -- type, which we need to constructed the error we really want.
+    -- Perhaps we do need to define our own set of errors.
+    -- _                      -> throwError (EExpectedRecordType ty)
```

Comment:
We generally format to be somewhat friendly to diffs. So no alignment, line breaks and indentation such that the amount of indentation does not depend on identifier length. Not a hard rule but I like being somewhat consistent here.